internal/pkg/api/plugin.go

```diff
@@ -0,0 +1,94 @@
+//      |  -----------       ---------   |   |  -----------                 -----------  |
+//      | | Collector | ==> |  Queue   | |==>| |  Filter   | ==>  ...  ==> |  Filter   | |
+//      | | (Parser)  |     | Mem/File | |   |  -----------                 -----------  |
+//      |  -----------       ----------  |   |      ||                          ||       |
+//       --------------------------------    |      \/	                        \/       |
+//                                           |  ---------------------------------------  |
+//                                           | |             OutputEventContext        | |
+//                                           |  ---------------------------------------  |
+//                                            -------------------------------------------
+//                             				   ||
+//                                             ||        --------------------------------
+//                                             ||     ->|         Sharing Client        |
+//                                             ||    |   --------------------------------
+//                                             ||    |
+//                                             \/    |            SegmentSender
+//                                            ---    |  ---------------------------------
+//                                           |   |   | |  -----------       -----------  |
+//                                           | D |   |-| |BatchBuffer| ==> | Forwarder | |
+//                                           | i |   | |  -----------       -----------  |
+//                                           | s |   |  ---------------------------------
+//                                           | p |   |
+//                                           | a |=> |              .......                 ===> Kafka/OAP
+//                                           | t |   |
+//                                           | c |   |             MeterSender
+//                                           | h |   | -----------------------------------
+//                                           | e |   -|  -------------       -----------  |
+//                                           | r |    | | BatchBuffer | ==> | Forwarder | |
+//                                           |   |    |  -------------       -----------  |
+//                                            ---      -----------------------------------
+//
+//
+// 1. The Collector plugin would fetch or receive the input data.
+// 2. The Parser plugin would parse the input data to SerializationEvent that is supported
+//    to be stored in Queue.
+// 3. The Queue plugin stores the SerializationEvent. However, the serialization depends on
+//    the Queue implements. For example, the serialization is unnecessary when using a Memory
+//    Queue.
+// 4. The Filter plugin would pull an event from the Queue and process the event to create
```

Comment:
I think consumer pull the data, filters are just processing data in a chain mode, right?

Reply:
Do you set up the event process one by one? I think the `one by one` mode is on the `filter`'s perspective only.

Reply:
right, would polish the comments